### PR TITLE
feat(ButtonIcon): New component for square button icons

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Summary:
New ButtonIcon component in 3 sizes - small, medium, large (to be used in Recommendations to begin with)

Screen Shot 2019-07-02 at 5 21 23 PM

ToDo:
- [x] Tokenize sizes

### CODE REVIEW

File: src/components/Icon/index.scss [hunk 1]
```diff
@@ -58,10 +58,10 @@
 /// Icon alignment fixes
 /// @description Fix for icons inside buttons with text and icons inside tabs.
 
-.#{$namespace}button .#{$namespace}icon,
-.oui-button:not(.oui-button--unstyled) .oui-icon {
-  vertical-align: sub;
-}
+// .#{$namespace}button .#{$namespace}icon,
```
Comment (ayeshakmaz) [inline src/components/Icon/index.scss]: @daverau-optimizely Wondering your thoughts on removing this, haven't yet found a case where this backfires by being deleted

File: src/main.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ export { default as EditableInput } from './components/EditableInput';
 export { default as EmptyDashboard } from './components/EmptyDashboard';
 export { default as FilterPicker } from './components/FilterPicker';
 export { default as HelpPopover } from './components/HelpPopover';
+export { default as ButtonIcon } from './components/ButtonIcon';
```
Comment (daverau-optimizely) [inline src/main.js]: Good memory! 😉

File: src/components/Icon/index.scss (hunk 1 above)
Comment (daverau-optimizely) [reply src/components/Icon/index.scss]: I'd be fairly certain this impacts something in the product but perhaps we should just remove it and see where it crops up. I'm for removing any CSS we can!

File: src/components/ButtonIcon/index.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
+.oui-button-icon {
```
Comment (daverau-optimizely) [inline src/components/ButtonIcon/index.scss]: Can we pair up on tokenizing these?

File: src/components/ButtonIcon/ButtonIcon.story.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import { storiesOf } from '@storybook/react';
+import { withKnobs, select, boolean } from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+
+import ButtonIcon from './index.js';
+import Container from '../Layout/Container';
```
Comment (daverau-optimizely) [inline src/components/ButtonIcon/ButtonIcon.story.js]: Fancy! You're the first consumer already 💯 🙌

**Pull request merged**